api/metrics/src/main/java/io/opentelemetry/api/metrics/BoundDoubleUpDownCounter.java

```diff
@@ -5,20 +5,29 @@
 
 package io.opentelemetry.api.metrics;
 
+import io.opentelemetry.context.Context;
 import javax.annotation.concurrent.ThreadSafe;
 
-/** A {@code Bound Instrument} for a {@link DoubleUpDownCounter}. */
+/** An up-down-counter instrument with pre-bound attributes. */
 @ThreadSafe
-public interface BoundDoubleUpDownCounter extends BoundSynchronousInstrument {
+public interface BoundDoubleUpDownCounter {
   /**
-   * Adds the given {@code increment} to the current value.
+   * Records a value with pre-bound attributes.
    *
-   * <p>The value added is associated with the current {@code Context}.
+   * <p>Note: This may use {@code Context.current()} to pull the context associated with this
+   * measurement.
    *
-   * @param increment the value to add.
+   * @param value The increment amount. May be positive, negative or zero.
    */
-  void add(double increment);
+  void add(double value);
+
+  /**
+   * Records a value with a pre-bound attributes.
+   *
+   * @param value The increment amount. May be positive, negative or zero.
+   * @param context The explicit context to associate with this measurement.
+   */
+  void add(double value, Context context);
 
-  @Override
   void unbind();
```

Comment:
I think all the `unbind()`s are going to need javadoc without the super interface.

Reply:
Added, copying previous javadoc.   Kinda unhappy with both bind + unbind's documentation on why/how they exists.

Thinking about discussing how (theoretically) if you only use `BOUND` instruments you should never see memory allocations in the hot path of your applications as, AFAIK, this is a guarantee from the metrics SDK.   However, this is just the API so that guarantee isn't assured, only possible.    WDYT?

Reply:
yeah, we can't guarantee this in the API javadoc, unfortunately.